Sluggo/View Controllers/LoginViewController.swift

```diff
@@ -12,66 +12,91 @@ class LoginViewController: UIViewController {
     }
 
     // This is also necessary when extending the superclass.
     required init?(coder aDecoder: NSCoder) {
-        self.config = Config()
-        self.token = "asdf" // will need to be changed
         super.init(coder: aDecoder)
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         self.isModalInPresentation = true
+        // only enable persistence button if the token is nil (that is, we didn't load an AppIdentity)
+        self.persistButton.isEnabled = (self.identity.token == nil)
+        self.loginButton.isEnabled = (self.identity.token == nil)
         persistButton.setImage(UIImage(systemName: "checkmark.square.fill"), for: [.highlighted, .selected])
     }
     
-    @IBAction func loginButton(_ sender: Any) {
-        
-        
-        // Sanitize input?
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         
+        if(identity.token != nil) {
+            // Segue out of VC
+            self.performSegue(withIdentifier: "loginToRoot", sender: self)
+        }
+    }
+    
+    @IBAction func loginButton(_ sender: Any) {
         let userString = username.text
         let passString = password.text
         
-        //print("User:", userString!)
-        //print("Password:", passString!)
-        
-        DispatchQueue.global(qos: .userInitiated).async {
-            self.loginMethod(userString!, passString!)
-        }
-        
         if(userString!.isEmpty || passString!.isEmpty) {
             // login Error
-            print("Login Error")
+            print("No username or password provided, not attempting login")
             return
         } else {
-            print("Login Success")
-            self.performSegue(withIdentifier: "loginToRoot", sender: self)
-            
+            DispatchQueue.global(qos: .userInitiated).async {
+                self.attemptLogin(username: userString!, password: passString!)
+            }
         }
     }
     
     
-    func loginMethod(_ user:String, _ password:String) {
-        let userManager = UserManager(config, token: "asdf")
-        userManager.doLogin(username: user, password: password) { result in
+    func attemptLogin(username: String, password: String) {
+        let userManager = UserManager(identity: identity)
+        userManager.doLogin(username: username, password: password) { result in
             switch(result) {
             case .success(let record):
-                print(record.key)
+                print("Successful login and retrieved token " + record.key)
+                
+                // Save to identity
+                self.identity.token = record.key
+                
+                // Persistence
+                // This is hacky but better than nothing
+                var rememberMe: Bool = false
+                DispatchQueue.main.sync {
+                    rememberMe = self.persistButton.isSelected
+                }
```

Comment:
Not really sure why the main.async is necessary. I think it would be better to capture the state of the persist button outside the callback.

Reply:
it probably would be better, but I don't think this has any real downsides, it works for now just looks kinda scuffed